app/src/main/java/com/mpieterse/stride/workers/PullWorker.kt

```
@@ -0,0 +1,64 @@
+            val checkinsSuccess = checkins.pull()
+            
+            if (habitsSuccess && checkinsSuccess) {
+                Clogger.d("PullWorker", "Successfully pulled all data from server")
+                Result.success()
+            } else {
+                Clogger.w("PullWorker", "Pull sync partially failed - habits: $habitsSuccess, checkins: $checkinsSuccess")
+                // Return success anyway - partial failures are handled in repositories
+                // and we don't want to retry immediately if one fails
+                Result.success()
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)